internal/controller/authenticator/webhookcachefiller/webhookcachefiller_test.go

```diff
@@ -1,96 +1,924 @@
+			LastTransitionTime: time,
+			Reason:             "InvalidEndpointURL",
+			Message:            fmt.Sprintf(`spec.endpoint URL is invalid: parse "%s": invalid character " " in host name`, issuer),
+		}
+	}
+	sadEndpointURLValidHTTPS := func(issuer string, time metav1.Time, observedGeneration int64) metav1.Condition {
+		return metav1.Condition{
+			Type:               "EndpointURLValid",
+			Status:             "False",
+			ObservedGeneration: observedGeneration,
+			LastTransitionTime: time,
+			Reason:             "InvalidEndpointURLScheme",
+			Message:            fmt.Sprintf(`spec.endpoint %s has invalid scheme, require 'https'`, issuer),
+		}
+	}
+
+	allHappyConditionsSuccess := func(endpoint string, someTime metav1.Time, observedGeneration int64) []metav1.Condition {
+		return conditionstestutil.SortByType([]metav1.Condition{
+			happyTLSConfigurationValidCAParsed(someTime, observedGeneration),
+			happyEndpointURLValid(someTime, observedGeneration),
+			happyTLSConnectionNegotiationValid(someTime, observedGeneration),
+			happyAuthenticatorValid(someTime, observedGeneration),
+			happyReadyCondition(someTime, observedGeneration),
+		})
+	}
+
+	webhookAuthenticatorGVR := schema.GroupVersionResource{
+		Group:    "authentication.concierge.pinniped.dev",
+		Version:  "v1alpha1",
+		Resource: "webhookauthenticators",
+	}
+	webhookAuthenticatorGVK := schema.GroupVersionKind{
+		Group:   "authentication.concierge.pinniped.dev",
+		Version: "v1alpha1",
+		Kind:    "WebhookAuthenticator",
+	}
+
 	tests := []struct {
 		name             string
 		syncKey          controllerlib.Key
 		webhooks         []runtime.Object
-		wantErr          string
-		wantLogs         []string
+		tlsDialerFunc    func(network string, addr string, config *tls.Config) (*tls.Conn, error)
+		wantSyncLoopErr  testutil.RequireErrorStringFunc
+		wantLogs         []map[string]any
+		wantActions      func() []coretesting.Action
 		wantCacheEntries int
 	}{
 		{
-			name:    "not found",
+			name:    "404: WebhookAuthenticator not found will abort sync loop, no status conditions",
+			syncKey: controllerlib.Key{Name: "test-name"},
+			wantLogs: []map[string]any{
+				{
+					"level":     "info",
+					"timestamp": "2099-08-08T13:57:36.123456Z",
+					"logger":    "webhookcachefiller-controller",
+					"message":   "Sync() found that the WebhookAuthenticator does not exist yet or was deleted",
+				},
+			},
+			wantActions: func() []coretesting.Action {
+				return []coretesting.Action{
+					coretesting.NewListAction(webhookAuthenticatorGVR, webhookAuthenticatorGVK, "", metav1.ListOptions{}),
+					coretesting.NewWatchAction(webhookAuthenticatorGVR, "", metav1.ListOptions{}),
+				}
+			},
+			wantCacheEntries: 0,
+		},
+		// Existing code that was never tested. We would likely have to create a server with bad clients to
+		// simulate this.
+		// { name: "non-404 `failed to get webhook authenticator` for other API server reasons" }
```

Comment:
Don't understand this comment. Was this a note to self?

Reply:
I thought I left a similar one on `JWTAuthenticator` just to document that non-404 api server errors aren't tested, but I see I did not.  I'll remove.